includes/user-functions.php

```diff
@@ -261,7 +262,7 @@ function give_count_donations_of_donor( $user = null ) {
 
 	$stats = ! empty( $user ) ? give_get_donation_stats_by_user( $user ) : false;
 
-	return isset( $stats['purchases'] ) ? $stats['purchases'] : 0;
+	return isset( $stats['donations_count'] ) ? $stats['donations_count'] : 0;
```

Comment:
If the donation history shortcode has a status param, you might want to account for that here so the pagination works for `any` or `completed/publish`

```suggestion
    return $status === 'any' ? ($stats['donations_count'] ?? 0) : ($stats['purchases'] ?? 0)
```